File: 백준/Silver/24479. 알고리즘 수업 － 깊이 우선 탐색 1/알고리즘 수업 － 깊이 우선 탐색 1.py
import sys
sys.setrecursionlimit(10**6)
input = sys.stdin.readline
def dfs(x):
  global cnt
  check[x] = cnt
  for i in a[x]:
    if not check[i]:
      cnt += 1
      dfs(i)
n,m,r = map(int,input().split())
a = [[] for _ in range(n+1)]
for _ in range(m):
  u,v = map(int,input().split())
  a[u].append(v)
  a[v].append(u)
check = [0]*(n+1)
cnt = 1
for i in range(n+1):
  a[i].sort()
dfs(r)
for i in range(1,n+1):
  print(check[i])

File: 백준/Silver/24479. 알고리즘 수업 － 깊이 우선 탐색 1/알고리즘 수업 － 깊이 우선 탐색 1.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
int n, m, r, cnt=1;
vector<vector<int>> a;
vector<int>check;

void dfs(int x) {
	check[x] = cnt;
	for (auto i : a[x]) {
		if (!check[i]) {
			cnt++;
			dfs(i);
		}
	}
}
int main() {
	cin >> n >> m >> r;
	a.resize(n + 1);
	check.resize(n + 1);
	for (int i = 0; i < m; i++) {
		int u, v;
		cin >> u >> v;
		a[u].push_back(v);
		a[v].push_back(u);
	}
	for (int i = 1; i <= n; i++) {
		sort(a[i].begin(), a[i].end());
	}
	dfs(r);
	for (int i = 1; i <= n; i++) {
		cout << check[i] << '\n';
	}
    return 0;
}
